### DISCUSSION TITLE
Question about exception handling of wasi function

### DISCUSSION THREAD

**Discussion by unicornt** [wasmerio/wasmer#4360] [General]
Hi, I noticed that `wasmer` handle error happened in wasi function different from `wasmtime`. For example, when meeting parameter that is not aligned to 4 in `fd_write`, `wasmtime` raises an exception `Pointer not aligned to 4` error while `wasmer` is not.
<img width="810" alt="image" src="https://github.com/wasmerio/wasmer/assets/18662046/f530b9c5-a518-4189-9cb7-2720e7312848">
I checked the specifications of wasi and found that the parameter was required to be aligned. 
![image](https://github.com/wasmerio/wasmer/assets/18662046/f66e5d0e-679c-4cd7-924b-b11832aaef39)
I'm quite curious about how you think about it.

**Comment by theduke** [comment]
I believe this currently happens to work on x86 because we don't specifically guard against it, but will fail on ARM architectures.

This working is more of an accident rather than a conscious design decision.
